drone/src/drone/agent/docker.rs

```diff
@@ -309,13 +316,29 @@ impl DockerInterface {
                         .collect(),
                     ),
                     runtime: self.runtime.clone(),
-                    // cpu_period: Some(DEFAULT_CPU_PERIOD),
-                    // cpu_quota: Some(DEFAULT_CPU_QUOTA),
-                    // ulimits: Some(vec![ResourcesUlimits {
-                    //     name: Some("cpu".to_string()),
-                    //     soft: Some(DEFAULT_CPU_TIME_ULIMIT),
-                    //     hard: Some(DEFAULT_CPU_TIME_ULIMIT),
-                    // }]),
+                    cpu_period: resource_limits
+                        .cpu_period
+                        .and_then(|cpu_period| Some(cpu_period.as_micros() as i64)),
```

Comment:
When the closure always returns a `Some`, `map()` is idiomatic because you don't have to `Some`-wrap the result, e.g.

```
cpu_period: resource_limits
                        .cpu_period
                        .map(|cpu_period| cpu_period.as_micros() as i64),
```